### PR TITLE
added docs for linchpin cli : needs revist in future

### PR DESCRIPTION
**Pull request by samvarankashyap**
Summary:
- fix typos in linchpin cli 
- added docs for linchpin cli

### CODE REVIEW

File: docs/source/linchpincli_linchpin_drop.rst [hunk 1]
```diff
@@ -0,0 +1,25 @@
...
+=========
+Examples:
+=========
+
++----------------------------+--------------------------------------------------------------------+
+| Usage                      | Action                                                             |
++----------------------------+--------------------------------------------------------------------+
+| linchpin drop              | Teardown  all targets in the PinFile of current working directory. |
++----------------------------+--------------------------------------------------------------------+
+| linchpin drop <targetname> | Initiates provisioning of specific target.                         |
```
Comment (herlo) [inline docs/source/linchpincli_linchpin_drop.rst]: This is inconsistent with the --target description above in the same file. Which should it be?

File: docs/source/linchpincli_linchpin_init.rst [hunk 2]
```diff
@@ -0,0 +1,20 @@
...
+  --help       Help
+
+=========
+Examples:
+=========
+
++------------------------+---------------------------------------------------------+
+| Usage                  | Action                                                  |
++========================+=========================================================+
+| linchpin init          |  initialises the directory structure for rise & drop    |
```
Comment (herlo) [inline docs/source/linchpincli_linchpin_init.rst]: This optionally creates a PinFile, prompting if the PinFile already exists. The docs should reflect this detail.

Comment (samvarankashyap) [inline docs/source/linchpincli_linchpin_init.rst]: @herlo added note mentioning the same

File: docs/source/linchpincli_linchpin_drop.rst (hunk 1 above)
Comment (samvarankashyap) [inline docs/source/linchpincli_linchpin_drop.rst]: @herlo currently it support --target only , made the change in following commit.

**Pull request merged**